### PR TITLE
Search parameters support for plugin

### PR DESCRIPTION
**Pull request by ajayns**
I've taken the case that searchParams is an object of parameters key-value pair, also handled the case where the value is an array. 

Ref: #62 

Next steps:

- [ ] Example use case in gatsby-starter-default

- [ ] Docs on search parameters, plugin-options for search params

### CODE REVIEW

File: src/gatsby-node.js [hunk 1]
```diff
@@ -20,6 +19,22 @@ const logMessage = (message, showLogs) => {
   }
 };
 
+// Helper to process an object of params
+// Converts to a string that can be added to a query
+const processParams = params =>
+  params
+    ? '?=' +
```
Comment (datakurre) [inline src/gatsby-node.js]: Why ?=

**Comment by datakurre** [comment]
yes

**Comment by ajayns** [comment]
@datakurre Is any refactoring possible that you'd recommend, I'm quite unsure if I'm following the best practices.

File: src/gatsby-node.js [hunk 2]
```diff
@@ -20,6 +19,23 @@ const logMessage = (message, showLogs) => {
...
+        .map(([key, value]) => {
+          if (Array.isArray(value)) {
+            return value.map(valueItem => `${key}=${valueItem}`).join();
+          } else {
+            return `${key}=${value}`;
+          }
+        })
+        .join()
+    : '';
+
```
Comment (datakurre) [inline src/gatsby-node.js]: After the first argument, the rest must be separated with &

url?arg=value&arg=value

Comment (datakurre) [reply src/gatsby-node.js]: Oh, arguments must be uriencoded.

**Comment by datakurre** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

**Comment by datakurre** [comment]
We are really in need of unit test setup for all this helper functions.

File: src/gatsby-node.js (hunk 2 above)
Comment (ajayns) [reply src/gatsby-node.js]: Oh damn, I'm so sorry, I should've done proper research and checked the code before pushing. This is my bad!

**Comment by ajayns** [comment]
Oh, this `encodeUri` thing, should be done for expansions as well then right? But that seems to be working fine without

**Comment by datakurre** [comment]
@ajayns expansion could also be escaped; they work because escaping (encodeuricomponent) is only required if parameter contains characters like & that have other meaning in url

It’s good practice anyway.

**Comment by ajayns** [comment]
So would it be a good practise to encode the expansions as well? 
@datakurre

**Comment by datakurre** [comment]
Yes. It might prevent issues where user tries to inject extra parameters within expansion value.

File: src/gatsby-node.js [hunk 3]
```diff
@@ -20,6 +19,27 @@ const logMessage = (message, showLogs) => {
   }
 };
 
+// Helper to process an object of params
+// Converts to a string that can be added to a query
+// Query format: '?key1=value1&key2=value2'
+const processParams = params => {
```
[the remark this answers is not in the document]
Comment (datakurre) [reply src/gatsby-node.js]: I did not know about axios and it’s features.

[the remark this answers is not in the document]
Comment (datakurre) [reply src/gatsby-node.js]: So also expands should be handled in the same way?

[the remark this answers is not in the document]
Comment (cekk) [reply src/gatsby-node.js]: yep..it's useful because we don't need to handle econdings

[the remark this answers is not in the document]
Comment (ajayns) [reply src/gatsby-node.js]: Oh wow, that looks good! I'll play around a bit and make the changes.

**Pull request merged**